ui/src/app/language-selector/language-selector.component.ts

```diff
@@ -11,17 +13,22 @@ import {SettingsService} from '../settings.service';
   styleUrls: ['./language-selector.component.scss']
 })
 export class LanguageSelectorComponent implements OnInit {
-  supportedLanguages: string[];
+  supportedLanguages: [LanguageCode, string][];
```

Comment:
This is a little clunky since the template has to magically refer to [0] or [1] depending on what value it wants. It may be better to define an interface LanguageOption with explicit displayString and languageCode properties

Reply:
done